README.md

```diff
@@ -49,6 +49,100 @@ This will set up an Electrode webapplication which will have 2 of the above 6 mo
   - [Electrode Confippet](https://github.com/electrode-io/electrode-confippet)
   - [Electrode Javascript Bundle Viewer](https://github.com/electrode-io/electrify)
 
+## Progressive Web App (PWA) features supported by the Electrode framework  
+#### 1. Offline first  
+  Offline first lets your app run without a network connection. At the same time it provides a great performance boost for repeat visit to your web site.
+  This is done with a service worker and by pre-caching your static assets as well as run time caching of react routes.  
+  [Learn More](https://github.com/electrode-io/electrode-archetype-react-app/blob/master/README.md#2-cache)  
+#### 2. Add To Homescreen  
+  After visiting you website, users will get a prompt to add your application to their homescreen (web or mobile). Combined with offline caching, this means your web app can be used exactly like a native application.  
+  [Learn More](https://github.com/electrode-io/electrode-archetype-react-app/blob/master/README.md#1-manifest)  
+#### 3. Push Notifications  
+  Web push notifications allow users to opt-in to timely updates from sites they love and allow you to effectively re-engage them with customized, relevant content.  
+  We will learn about Push Notifications in the next couple of sections.
+
+## Instructions for setting up push notification
+Push Notification is based on service workers because service workers operate in the background. So we need to register our service worker first.  
+Check out [this guideline](https://github.com/electrode-io/electrode-archetype-react-app/blob/master/README.md#how-do-i-generate-a-manifestjson-and-a-service-worker-file) to generate a service worker in an electrode app.  
+
+Next we need to add a `push` event to this existing service worker:  
+#### 1. Create a new file `sw.js`,  
+```
+self.addEventListener("push", (event) => {
+  const title = "It worked!";
+  const options = {
+    body: "Great job sending that push notification!"
+  };
+  event.waitUntil(
+    self.registration.showNotification(title, options)
+  );
+});
+```
+[Sample file](https://github.com/electrode-io/electrode-boilerplate-universal-react-node/blob/master/client/sw.js)
+
+
+#### 2. Include this file in your webpack bundle by referencing it in `sw-config.js` :
+```
+module.exports = {
+  cache: {
+    importScripts: ['./sw.js']
+    }
+}
+```
+[Sample file](https://github.com/electrode-io/electrode-boilerplate-universal-react-node/blob/master/config/sw-config.js)
+
+
+#### 3. Now we can register this `push` event in the browser and install the new service worker:  
```

Comment:
Our service worker registrations works with the file generated with sw-config. I was under the impression that we are doing the re registration to add the push event?